src/relayclient/ContractInteractor.ts

```diff
@@ -259,31 +259,64 @@ export default class ContractInteractor {
     return latestBlock.gasLimit
   }
 
-  async validateAcceptRelayCall (
+  /**
+   * make a view call to relayCall(), just like the way it will be called by the relayer.
+   * returns:
+   * - paymasterAccepted - true if accepted
+   * - reverted - true if relayCall was reverted.
+   * - returnValue - if either reverted or paymaster NOT accepted, then this is the reason string.
+   */
+  async validateRelayCall (
     paymasterMaxAcceptanceBudget: number,
     relayRequest: RelayRequest,
     signature: PrefixedHexString,
     approvalData: PrefixedHexString): Promise<{ paymasterAccepted: boolean, returnValue: string, reverted: boolean }> {
     const relayHub = this.relayHubInstance
     try {
       const externalGasLimit = await this._getBlockGasLimit()
-
-      const res = await relayHub.contract.methods.relayCall(
+      const encodedRelayCall = relayHub.contract.methods.relayCall(
         paymasterMaxAcceptanceBudget,
         relayRequest,
         signature,
         approvalData,
         externalGasLimit
-      )
-        .call({
-          from: relayRequest.relayData.relayWorker,
-          gasPrice: relayRequest.relayData.gasPrice,
-          gas: externalGasLimit
+      ).encodeABI()
+      const res = await new Promise((resolve, reject) => {
+        // @ts-ignore
+        this.web3.currentProvider.send({
+          jsonrpc: '2.0',
+          id: 1,
+          method: 'eth_call',
+          params: [
+            {
+              from: relayRequest.relayData.relayWorker,
+              to: relayHub.address,
+              gasPrice: toHex(relayRequest.relayData.gasPrice),
+              gas: toHex(externalGasLimit),
+              data: encodedRelayCall
+            },
+            'latest'
+          ]
+        }, (err: any, res: { result: string }) => {
+          const revertMsg = this._decodeRevertFromResponse(err, res)
+          if (revertMsg != null) {
+            reject(new Error(revertMsg))
+          }
+          if (err !== null) {
+            reject(err)
+          } else {
+            resolve(res.result)
+          }
         })
-      log.info(res)
+      })
+
+      // @ts-ignore
+      const decoded = abi.decodeParameters(['bool', 'string'], res as string)
```

Comment:
I think you can do `const res: string = await blah()`, so you don't need to cast every time you use it

Reply:
ok